### PR TITLE
Added Handling for Creating, Updating and Deleting Resources

### PR DESCRIPTION
**Pull request by mcuyar**
- Add request middleware w/injecting a new ApiRequest
- Added resolvers for returning the api request singleton for both Illuminate Reqests and ApiRequest
- Added a manageRelationship trait for handling relationships on update and create
- Updated repositories for handling requests dynamically
- Updated Generated controllers and repositories
- Upated unit tests to reflect changes

### CODE REVIEW

File: luminary/Services/ApiRequest/Traits/HasForbiddenAttribute.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Luminary\Services\ApiRequest\Traits;
+
+use Luminary\Services\ApiRequest\Exceptions\ForbiddenAttribute;
+
+trait HasForbiddenAttribute
+{
+
```
Comment (bluewebtech) [inline luminary/Services/ApiRequest/Traits/HasForbiddenAttribute.php]: Remove blank indentation line.

File: tests/ApiRequest/ApiRequestSetMiddlewareTest.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+        app()->post('/api', function(Request $request, \Luminary\Services\Testing\Requests\FormRequest $formRequest, \Luminary\Services\ApiRequest\ApiRequest $apiRequest) use($self) {
+            $self->assertInstanceOf(\Luminary\Services\ApiRequest\ApiRequest::class, $request);
+            $self->assertInstanceOf(\Luminary\Services\ApiRequest\ApiRequest::class, $formRequest);
+            $self->assertInstanceOf(\Luminary\Services\ApiRequest\ApiRequest::class, $apiRequest);
+        });
+
+        $this->json('post', '/api', [
+            'data' => [
+
```
Comment (bluewebtech) [inline tests/ApiRequest/ApiRequestSetMiddlewareTest.php]: Remove blank indentation line.

File: tests/Generators/GeneratedControllersTest.php [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+        $data = [
+            'data' => [
+                'type' => 'customers',
+                'id' => $id,
+                'attributes' => $attributes
+            ]
+        ];
+
+        $this->json('PATCH', 'customers/'.$id, $data , $this->headers());
```
Comment (bluewebtech) [inline tests/Generators/GeneratedControllersTest.php]: Remove space after $data variable within arg list.

File: tests/Generators/GeneratedControllersTest.php [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+    /**
+     * Test that the Delete method returns the correct response
+     *
+     * @return void
+     */
+    public function testDeleteMethod() :void
+    {
+        $id = $this->customers->first()->id;
+
+        $this->json('DELETE', 'customers/'.$id, [] , $this->headers());
```
Comment (bluewebtech) [inline tests/Generators/GeneratedControllersTest.php]: Remove space after [] within arg list.

File: tests/Generators/GeneratedRepositoriesTest.php [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+    /**
+     * Setup the test environment.
+     *
+     * @return void
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->seed(5,5,5);
```
Comment (bluewebtech) [inline tests/Generators/GeneratedRepositoriesTest.php]: Space out values in arg list.

**Comment by bluewebtech** [comment]
Nothing crazy, only minor formatting to correct.

**Comment by mcuyar** [comment]
@bluewebtech Updated commit with changes.

**Pull request merged**